src/Traits/HasPermissions.php

```diff
@@ -117,8 +117,9 @@ protected function getGuardNames(): Collection
     protected function getDefaultGuardName(): string
     {
         $default = config('auth.defaults.guard');
+        $guardNames = $this->getGuardNames();
 
-        return $default ?: $this->getGuardNames()->first();
+        return $guardNames->contains($default) ? $default : $this->getGuardNames()->first();
```

Comment:
Would this be more performant to call `$guardNames->first()` instead of rerunning `$this->getGuardNames()` a second time?

(I'm still not sure that I agree with the overall proposed logic change, but if it is to go ahead, it feels like it should be optimized.)

Reply:
You are right, it's an oversight. I'll change it in this PR